lib/modules/datasource/docker/common.ts

```diff
@@ -99,6 +100,23 @@ export async function getAuthHeaders(
       if (auth) {
         opts.headers = { authorization: `Basic ${auth}` };
       }
+    } else if (
+      googleRegex.test(registryHost) &&
+      typeof opts.username === 'undefined' &&
+      typeof opts.password === 'undefined' &&
+      typeof opts.token === 'undefined'
+    ) {
+      logger.trace(
+        { registryHost, dockerRepository },
+        `Using google auth for Docker registry`
+      );
+      const accessToken = await getGoogleAccessToken();
+      const auth = Buffer.from(`${"oauth2accesstoken"}:${accessToken}`).toString(
+        'base64'
+      );
+      if (auth) {
+        opts.headers = { authorization: `Basic ${auth}` };
+      }
```

Comment:
```suggestion
      opts.headers = { authorization: `Basic ${auth}` };
```
auth should never be null or undefined here